golem-component-service-base/src/service/component.rs

```diff
@@ -920,94 +898,10 @@ impl<Owner: ComponentOwner, Scope: PluginScope> ComponentServiceDefault<Owner, S
-                                )
-                            },
-                        )?)
-                        .mime_str("application/json")
-                        .unwrap(),
-                    );
-                    for (key, value) in *parameters {
-                        if key == "component" {
-                            return Err(ComponentError::TransformationFailed(
-                                TransformationFailedReason::Failure(
-                                    "Parameter key 'component' is reserved".to_string(),
-                                ),
-                            ));
-                        }
-                        if key == "metadata" {
-                            return Err(ComponentError::TransformationFailed(
-                                TransformationFailedReason::Failure(
-                                    "Parameter key 'metadata' is reserved".to_string(),
-                                ),
-                            ));
-                        }
-                        form = form.part(key.clone(), Part::text(value.clone()));
-                    }
-
-                    let request = client.post(url).multipart(form);
-
-                    let response = request.send().await.map_err(|err| {
-                        ComponentError::TransformationFailed(TransformationFailedReason::Request(
-                            err,
-                        ))
-                    })?;
-
-                    if response.status().is_server_error() {
-                        return Err(ComponentError::TransformationFailed(
-                            TransformationFailedReason::HttpStatus(response.status()),
-                        ));
-                    }
-
-                    Ok(response)
-                })
-            },
-            |err| {
-                matches!(
-                    err,
-                    ComponentError::TransformationFailed(TransformationFailedReason::HttpStatus(_))
-                        | ComponentError::TransformationFailed(
-                            TransformationFailedReason::Request(_)
-                        )
-                )
-            },
-        )
-        .await?;
-
-        if response.status().is_success() {
-            let body = response.bytes().await.map_err(|err| {
-                ComponentError::TransformationFailed(TransformationFailedReason::Failure(format!(
-                    "Failed to read response from transformation plugin: {}",
-                    err
-                )))
-            })?;
-
-            Ok(body.to_vec())
-        } else {
-            Err(ComponentError::TransformationFailed(
-                TransformationFailedReason::HttpStatus(response.status()),
-            ))
-        }
+        self.transformer_plugin_caller
+            .call_remote_transformer_plugin(component, data, url, parameters)
+            .await
+            .map_err(ComponentError::TransformationFailed)
```

Comment:
Decided to not go with 502 here, as it would be ambiguous with alb errors. Added a check that at least asserts that we have an error message exposing the downstream failure.